pkg/genericapiserver/genericapiserver.go

```diff
@@ -269,6 +269,10 @@ type GenericAPIServer struct {
 	enableOpenAPISupport   bool
 	openAPIInfo            spec.Info
 	openAPIDefaultResponse spec.Response
+
+	// PostStartHooks is called in a go func after the server has started listening.
+	// It may kill the process with a panic if it wishes to by returning an error
+	PostStartHooks []func() error
```

Comment:
I do not like this factoring.
* For debugability, funcs need a name, so we can say what went wrong when they fail.
* Funcs should have a named type.
* They should accept some sort of PostStartContext type which is a struct, so we can add fields to it without changing the entire codebase